test/metabase/query_processor_test/nested_queries_test.clj

```diff
@@ -147,44 +147,49 @@
+  (mt/test-drivers (mt/normal-drivers-with-feature :nested-queries)
+    (mt/dataset sample-dataset
+      (doseq  [dataset? [true false]]
+        (testing (format "Aggregations in both nested and outer query for %s have correct metadata (#19403) and (#23248)"
+                         (if dataset? "questions" "models"))
+          (mt/with-temp* [Card [{card-id :id :as card}
+                                {:dataset dataset?
+                                 :dataset_query
+                                 (mt/$ids :products
+                                          {:type     :query
+                                           :database (mt/id)
+                                           :query    {:source-table $$products
+                                                      :aggregation
+                                                      [[:aggregation-options
+                                                        [:sum $price]
+                                                        {:name "sum"}]
+                                                       [:aggregation-options
+                                                        [:max $rating]
+                                                        {:name "max"}]]
+                                                      :breakout     $category
+                                                      :order-by     [[:asc $category]]}})}]]
+            (is (= {:cols [{:name "sum" :display_name "Sum of Sum of Price"}
+                           {:name "count" :display_name "Count"}]
+                    :rows [[11149 4]]}
+                   (-> (mt/format-rows-by [int int]
+                         (qp/process-query (merge {:type     :query
+                                                   :database (mt/id)
+                                                   :query    {:source-table (str "card__" card-id)
+                                                              :aggregation  [[:aggregation-options
+                                                                              [:sum
+                                                                               [:field
+                                                                                "sum"
+                                                                                {:base-type :type/Float}]]
+                                                                              {:name "sum"}]
+                                                                             [:aggregation-options
+                                                                              [:count]
+                                                                              {:name "count"}]]}}
+                                                  (when dataset?
+                                                    {:info {:metadata/dataset-metadata (:result_metadata card)}}))))
+                       :data
```

Comment:
Instead of all the `:data` and `select-keys` and stuff here you should rework this test to use `partial=`. Why? Because if the query errors there will be no `:data` key. And debugging it is going to be a pain in the butt. With `partial=` if the query fails the test failure will print the entire map including the error message.